### PR TITLE
Open3d library usage

### PR DESCRIPTION
**Pull request by grigorovici-andreea**
Implemented bindings to open3d in python and cpp.
Solves issue: #85

### CODE REVIEW

**Comment by dNechita** [comment]
Add the issue that this PR is addressing in the description section.

File: bindings/open3D/aditof_open3d.h [hunk 1]
```diff
@@ -0,0 +1,64 @@
+#ifndef ADITOF_OPENCV_H
```
Comment (dNechita) [inline bindings/open3D/aditof_open3d.h]: The include guards are from another file.

File: bindings/open3D/aditof_open3d.h [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+#include <Open3D/Open3D.h>
+#include <aditof/frame.h>
+#include <aditof/status_definitions.h>
+
+using namespace open3d;
+
+namespace aditof {
+aditof::Status fromFrameToDepthImg(aditof::Frame &frame, int camera_range,
+                                   geometry::Image &image) {
+    aditof::FrameDetails frameDetails;
```
Comment (dNechita) [inline bindings/open3D/aditof_open3d.h]: This function is defined inside the aditof namespace, no need to add the namespace to FrameDetails. Check the rest of the file for more issues like this.

File: bindings/open3D/aditof_open3d.h [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    uint16_t *depthData;
+    frame.getData(aditof::FrameDataType::DEPTH, &depthData);
+
+    if (depthData == nullptr) {
+        return aditof::Status::GENERIC_ERROR;
+    }
+
+    image.Prepare(frameWidth, frameHeight, 1, 1);
+    for (int i = 0; i < frameHeight * frameWidth; i++) {
+        uint8_t *p = (uint8_t *)(image.data_.data() + i * sizeof(uint8_t));
```
Comment (dNechita) [inline bindings/open3D/aditof_open3d.h]: old cast

File: bindings/open3D/showPointCloud/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,16 @@
+cmake_minimum_required(VERSION 2.8)
+
+project(aditof-open3d-showPointCloud)
+
+message(STATUS "affafaaaf Open3D ${Open3D_VERSION}")
```
Comment (dNechita) [inline bindings/open3D/showPointCloud/CMakeLists.txt]: should be removed

File: bindings/open3D/showPointCloud/main.cpp [hunk 5]
```diff
@@ -0,0 +1,138 @@
+#include <aditof/camera.h>
+#include <aditof/frame.h>
+#include <aditof/system.h>
+#include <glog/logging.h>
+
+#include "../aditof_open3d.h"
+
+static const uint8_t colormap[3 * 256] = {
+#include "colormap.txt"
```
Comment (dNechita) [inline bindings/open3D/showPointCloud/main.cpp]: This doesn't look right

File: bindings/open3D/showPointCloud/main.cpp [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+    Status status = Status::OK;
+
+    System system;
+    status = system.initialize();
+    if (status != Status::OK) {
+        LOG(ERROR) << "Could not initialize system!";
+        return 0;
+    }
+
+    std::vector<Camera *> cameras;
```
Comment (dNechita) [inline bindings/open3D/showPointCloud/main.cpp]: getCameraList() provides shared pointers now

File: bindings/open3D/showPointCloud/main.cpp [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+    if (status != Status::OK) {
+        LOG(ERROR) << "Could not request frame!";
+        return 0;
+    }
+    aditof::FrameDetails frameDetails;
+    frame.getDetails(frameDetails);
+    int frameHeight = static_cast<int>(frameDetails.height) / 2;
+    int frameWidth = static_cast<int>(frameDetails.width);
+
+    //    auto Vimage = std::make_shared<geometry::Image>();
```
Comment (dNechita) [inline bindings/open3D/showPointCloud/main.cpp]: still required?

File: bindings/open3D/showPointCloud/main.cpp [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+        if (status != Status::OK) {
+            LOG(ERROR) << "Could not request frame!";
+            return 0;
+        }
+        geometry::Image depth_image;
+        status = fromFrameToDepthImg(frame, camera_range, depth_image);
+        if (status != Status::OK) {
+            LOG(ERROR) << "Could not convert from frame to Image!";
+        }
+        //        memcpy(Vimage->data_.data(), depth_image.data_.data(),
```
Comment (dNechita) [inline bindings/open3D/showPointCloud/main.cpp]: required?

File: bindings/open3D/showPointCloud/main.cpp [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+                   &colormap[depth_image.data_[i] * 3], 3);
+        }
+        io::WriteImage("color.png", color_image);
+        camera::PinholeCameraIntrinsic camera;
+        auto rgbd_ptr = geometry::RGBDImage::CreateFromColorAndDepth(
+            color_image, depth_image, false);
+        auto pointcloud_ptr =
+            geometry::PointCloud::CreateFromRGBDImage(*rgbd_ptr, camera);
+
+        LOG(ERROR) << "AAAAA";
```
Comment (dNechita) [inline bindings/open3D/showPointCloud/main.cpp]: Please remove

File: bindings/python/examples/open3d/tryout.py [hunk 10]
```diff
@@ -0,0 +1,137 @@
+import aditofpython as tof
```
[the remark this answers is not in the document]
Comment (grigorovici-andreea) [reply bindings/python/examples/open3d/tryout.py]: fixed. thanks

File: bindings/python/examples/open3d/tryout.py [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+
+    device = cameras[0].getDevice()
+
+    smallSignalThreshold = 50;
+    REGS_CNT = 5
+    afeRegsAddr = [0x4001, 0x7c22, 0xc34a, 0x4001,
+                   0x7c22]
+    afeRegsVal = [0x0006, 0x0004, 0x8032, 0x0007, 0x0004]
+
+    # I tried instead of 0, 0x8000
```
[the remark this answers is not in the document]
Comment (grigorovici-andreea) [reply bindings/python/examples/open3d/tryout.py]: it is already removed. thanks

File: bindings/python/examples/open3d/tryout.py [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+        print("cameras[0].setMode() failed with status: ", status)
+
+    camDetails = tof.CameraDetails()
+    status = cameras[0].getDetails(camDetails)
+    if not status:
+        print("system.getDetails() failed with status: ", status)
+
+    device = cameras[0].getDevice()
+
+    smallSignalThreshold = 50;
```
[the remark this answers is not in the document]
Comment (grigorovici-andreea) [reply bindings/python/examples/open3d/tryout.py]: solved. thanks

File: bindings/open3D/CMakeLists.txt [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+
+# Set OS-specific things here
+if(WIN32)
+elseif(CYGWIN)
+elseif(APPLE)
+elseif(UNIX)
+	add_definitions(-DUNIX)
+	add_compile_options(-Wno-deprecated-declarations)
+	add_compile_options(-Wno-unused-result)
+    add_definitions(-O3)
```
[the remark this answers is not in the document]
Comment (grigorovici-andreea) [reply bindings/open3D/CMakeLists.txt]: solved. (deleted the unnecessary definitions too) thanks.

File: bindings/open3D/aditof_open3d.h (hunk 2 above)
Comment (grigorovici-andreea) [reply bindings/open3D/aditof_open3d.h]: solved. thanks

File: bindings/open3D/aditof_open3d.h (hunk 3 above)
Comment (grigorovici-andreea) [reply bindings/open3D/aditof_open3d.h]: solved. thanks

File: bindings/open3D/CMakeLists.txt [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+    list(APPEND Open3D_LIBRARIES dl)
+endif()
+
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${Open3D_C_FLAGS}")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${Open3D_CXX_FLAGS}")
+set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${Open3D_EXE_LINKER_FLAGS}")
+
+# Set OS-specific things here
+if(WIN32)
+elseif(CYGWIN)
```
[the remark this answers is not in the document]
Comment (grigorovici-andreea) [reply bindings/open3D/CMakeLists.txt]: solved. thanks

**Pull request merged**